### PR TITLE
Use global logger for all application logs

### PR DESCRIPTION
**Pull request by ffilippopoulos**
Borrow the logger from wireguard device package to bring consistency across
application and tun devices logs

### CODE REVIEW

File: agent.go [hunk 1]
```diff
@@ -32,15 +31,20 @@ func NewAgent(cfg *agentConfig) *Agent {
...
 			continue
 		}
 		agent.deviceManagers = append(agent.deviceManagers, dm)
 	}
 	tokenDir := filepath.Dir(defaultTokenFileLoc)
 	err := os.MkdirAll(tokenDir, 0755)
 	if err != nil {
-		log.Printf("Error: unable to create directory=%s", tokenDir)
+		logger.Error.Printf(
+			"Error: unable to create directory=%s", tokenDir)
```
Comment (george-angel) [inline agent.go:47]: Do you still need `Error:` if its part of the logging object now?

**Review by george-angel — APPROVED** [review]

File: agent.go (hunk 1 above)
Comment (ffilippopoulos) [reply agent.go:47]: good catch!

Comment (ffilippopoulos) [reply agent.go]: fixed

File: devicemanager_darwin.go [hunk 2]
```diff
@@ -45,19 +46,28 @@ func (dm *DeviceManager) updateDeviceConfig(oldConfig, config *WirestewardPeerCo
 		// routes via interfaces.
 		for _, r := range oldConfig.AllowedIPs {
 			if err := delRoute(fdRoute, oldConfig.LocalAddress.IP, r.IP, r.Mask); err != nil {
-				log.Printf("Could not remove old route (%s): %s", r, err)
+				logger.ErrorPrintf(
```
Comment (alkar) [inline devicemanager_darwin.go:49]: Missing a dot here.

File: logger.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		logLevel = device.LogLevelDebug
+	case "info":
+		logLevel = device.LogLevelInfo
+	case "error":
+		logLevel = device.LogLevelError
+	}
+}
+
+// Returns a new logger using the global level variable
+func initLogger(name string) *device.Logger {
```
Comment (alkar) [inline logger.go:29]: Since this is returning a new `device.Logger` maybe we should call it `newLogger` instead?

File: logger.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+// initialisation of loggers
+func setLogLevel(level string) {
+	switch level {
+	case "debug":
+		logLevel = device.LogLevelDebug
+	case "info":
+		logLevel = device.LogLevelInfo
+	case "error":
+		logLevel = device.LogLevelError
+	}
```
Comment (alkar) [inline logger.go:25]: Do we want to catch an invalid value here and notify the user?
Silently ignoring a wrong value and using the default level isn't too bad in my opinion, but I thought I'd mention this.

Comment (ffilippopoulos) [reply logger.go:25]: ouch, darwin! that's why I didn't catch this on build/test, thanks!

**Review by alkar — APPROVED** [review]

**Review by alkar — APPROVED** [review]

**Review by alkar — APPROVED** [review]

**Pull request merged**